library/src/main/java/com/getstream/sdk/chat/rest/core/providers/WebSocketServiceProvider.java

```diff
@@ -0,0 +1,70 @@
+package com.getstream.sdk.chat.rest.core.providers;
+
+import android.util.Log;
+
+import com.getstream.sdk.chat.interfaces.WSResponseHandler;
+import com.getstream.sdk.chat.rest.User;
+import com.getstream.sdk.chat.rest.WebSocketService;
+import com.getstream.sdk.chat.rest.core.ApiClientOptions;
+import com.getstream.sdk.chat.rest.core.Client;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+/*
+ * Created by Anton Bevza on 2019-10-24.
+ */
+public class WebSocketServiceProvider {
```

Comment:
These providers should be already replaced with interfaces. So in the end constructor of `Client` should accept only interfaces and maybe primitive config values.